### PR TITLE
🐛 fixed hooks warning on the AddEmission screen

### LINKED ISSUE #156: Fix hooks warning
Fix hooks warning in AddEmission screen :

hooks

### PR DESCRIPTION
**Pull request by paulussujono**
closes and fixes #156

- added the recommended react hooks eslint rules
- fixed the hooks order warning on the AddEmission screen
- fixed some other existing eslint warnings

### CODE REVIEW

File: app/screens/AddEmission/AddEmissionScreen.navigationOptions.tsx [hunk 1]
```diff
@@ -1,33 +1,18 @@
...
-import { useNavigation } from "@react-navigation/native";
 
 import { Text } from "../../components";
 import { t } from "../../utils";
 import { Colors, ComponentsStyle } from "../../style";
-import { navigate } from "../../navigation";
+import { InfoButton } from "./components";
 
-const navigationOptions = () => {
-  const navigation = useNavigation();
```
Comment (paulussujono) [inline app/screens/AddEmission/AddEmissionScreen.navigationOptions.tsx:12]: after adding the recommended hook rules lint rule, this hook call was shown to be used outside of a React component, which caused the warnings. I have extracted the hook call and its usage into a separate component.

File: app/screens/Settings/SettingsScreen.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import {
   ScrollView,
   Image,
 } from "react-native";
-import Constants from "expo-constants";
```
Comment (paulussujono) [inline app/screens/Settings/SettingsScreen.tsx:8]: fixed existing eslint warning for [import/no-named-as-default](https://github.com/benmosher/eslint-plugin-import/blob/v2.22.1/docs/rules/no-named-as-default.md)
`Using exported name 'Constants' as identifier for default export`

File: app/screens/Splash/SplashScreen.tsx [hunk 3]
```diff
@@ -7,18 +7,17 @@ interface Props {
...
     Animated.timing(fadeAnim, {
       toValue: 1,
       duration: 3000,
       useNativeDriver: false,
     }).start(({ finished }) => {
-      props.screenAnimationComplete(finished);
+      screenAnimationComplete(finished);
     });
-  }, [fadeAnim]);
+  }, [fadeAnim, screenAnimationComplete]);
```
Comment (paulussujono) [inline app/screens/Splash/SplashScreen.tsx:20]: rules of hooks - missing dependency

File: app/screens/Splash/SplashScreen.tsx [hunk 4]
```diff
@@ -7,18 +7,17 @@ interface Props {
   screenAnimationComplete: (boolean) => void;
 }
 
-const SplashScreen = (props: Props) => {
-  const fadeAnim = new Animated.Value(0);
-
+const SplashScreen = ({ screenAnimationComplete }: Props) => {
+  const fadeAnim = useMemo(() => new Animated.Value(0), []);
```
Comment (paulussujono) [inline app/screens/Splash/SplashScreen.tsx:11]: fixing `eslintreact-hooks/exhaustive-deps`

```The 'fadeAnim' object construction makes the dependencies of useEffect Hook (at line 21) change on every render. To fix this, wrap the initialization of 'fadeAnim' in its own useMemo() Hook.```

**Comment by PierreBresson** [comment]
Looks good, thank you!

**Pull request merged**